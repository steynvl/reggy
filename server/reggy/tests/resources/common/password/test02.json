{
  "type": "Password",
  "information": {
    "shouldStartWith": "Letter",
    "shouldContain": [
      "Digit",
      "Special character",
      "Minus"
    ],
    "minimumLength": "4",
    "maximumLength": "22"
  },
  "generalRegexInfo": {
    "startRegexMatchAt": "Anywhere",
    "endRegexMatchAt": "Anywhere",
    "regexTarget": "Java"
  },
  "generateMethod": "commonUseCases"
}