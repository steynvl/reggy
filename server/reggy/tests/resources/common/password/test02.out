Java
Pattern regex = Pattern.compile("[A-Za-z](?=.*\\d)(?=.*[!\"#$%&'()*+,-.:;<=>?@\\[\\]^_`{|}~])(?=.*-).{3,21}");
[A-Za-z](?=.*\\d)(?=.*[!\"#$%&'()*+,-.:;<=>?@\\[\\]^_`{|}~])(?=.*-).{3,21}
Perl
my $regex = /[A-Za-z](?=.*\d)(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~])(?=.*-).{3,21}/;
[A-Za-z](?=.*\d)(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~])(?=.*-).{3,21}
POSIX
[A-Za-z](?=.*\d)(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~])(?=.*-).{3,21}
[A-Za-z](?=.*\d)(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~])(?=.*-).{3,21}
Python
regex = re.compile(r'[A-Za-z](?=.*\d)(?=.*[!"#$%&\'()*+,-.:;<=>?@\[\]^_`{|}~])(?=.*-).{3,21}')
[A-Za-z](?=.*\d)(?=.*[!"#$%&\'()*+,-.:;<=>?@\[\]^_`{|}~])(?=.*-).{3,21}
JavaScript
const regex = /[A-Za-z](?=.*\d)(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~])(?=.*-).{3,21}/;
[A-Za-z](?=.*\d)(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~])(?=.*-).{3,21}
PHP
$regex = '/[A-Za-z](?=.*\d)(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~])(?=.*-).{3,21}/';
[A-Za-z](?=.*\d)(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~])(?=.*-).{3,21}
Golang
regex, _ := regexp.Compile("[A-Za-z](?=.*\\d)(?=.*[!\"#$%&'()*+,-.:;<=>?@\\[\\]^_`{|}~])(?=.*-).{3,21}")
[A-Za-z](?=.*\\d)(?=.*[!\"#$%&'()*+,-.:;<=>?@\\[\\]^_`{|}~])(?=.*-).{3,21}
Rust
let re = Regex::new(r"[A-Za-z](?=.*\d)(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~])(?=.*-).{3,21}").unwrap();
[A-Za-z](?=.*\d)(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~])(?=.*-).{3,21}
C#
var re = new Regex(@"[A-Za-z](?=.*\d)(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~])(?=.*-).{3,21}");
[A-Za-z](?=.*\d)(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~])(?=.*-).{3,21}
Scala
val re = "[A-Za-z](?=.*\\d)(?=.*[!\"#$%&'()*+,-.:;<=>?@\\[\\]^_`{|}~])(?=.*-).{3,21}".r
[A-Za-z](?=.*\\d)(?=.*[!\"#$%&'()*+,-.:;<=>?@\\[\\]^_`{|}~])(?=.*-).{3,21}
Kotlin
val regex = Regex("[A-Za-z](?=.*\\d)(?=.*[!\"#$%&'()*+,-.:;<=>?@\\[\\]^_`{|}~])(?=.*-).{3,21}")
[A-Za-z](?=.*\\d)(?=.*[!\"#$%&'()*+,-.:;<=>?@\\[\\]^_`{|}~])(?=.*-).{3,21}